### ISSUE TITLE
Exposure tuning (nightpreview)

### ISSUE THREAD

**Issue by nopmop** [kerberos-io/machinery#192]
**Problem**
my problem is **dark images** - I'm aware of the FPS<->light relationship (as detailed at https://github.com/kerberos-io/machinery/issues/12) and of possible workarounds (like using an RTSP server as detailed here https://github.com/kerberos-io/machinery/issues/165, which hammers the CPU considerably though).

**Solution i'd like**
Being able to tune the `mmal exposure` parameter (set to `nightpreview` for example) from kerberosio would be nice. 

**Question**
Is this going to be contemplated for a future release? Where to look in the code to submit a PR?

**Comment by nopmop** [comment]
Bump. Any help?

**Comment by cedricve** [comment]
hey @nopmop, this is a hard one. You will need to consider looking into this library.github.com/dridri/OpenMaxIL-cpp

**Comment by gbus** [comment]
Hi @cedricve 
I have just added support for setExposureControl in https://github.com/dridri/OpenMaxIL-cpp.
I have tested it from kerberos.io with this simple change, but I guess that it takes you 5 minutes to make it configurable from the raspicam settings in capture.xml:

```
diff --git a/src/kerberos/capture/RaspiCamera.cpp b/src/kerberos/capture/RaspiCamera.cpp
index 07c1a97..f17ef9c 100755
--- a/src/kerberos/capture/RaspiCamera.cpp
+++ b/src/kerberos/capture/RaspiCamera.cpp
@@ -218,7 +218,7 @@ namespace kerberos
 				state.camera->setSaturation(m_saturation);
 				state.camera->setContrast(m_contrast);
 				state.camera->setSharpness(m_sharpness);
-				//state.camera->setExposureControl(Camera::ExposureControlOff);
+				state.camera->setExposureControl(Camera::ExposureControlNight);
 
 				if(m_angle > 0)
 				{
```

Thanks

**Comment by cedricve** [comment]
super cool @gbus, do you have any results already?

**Comment by gbus** [comment]
Yes, with ExposureControlNight the image from a room with little light looked very bright, while with ExposureControlSnow was totally dark.
:-)

**Comment by cedricve** [comment]
Cool ! and what about the recording speed, just curious. How much FPS do you get.
Will definitely check this out myself. Great job!!

**Comment by gbus** [comment]
I tried 1,3,5,10,20,30 fps, where at 10fps the exposure was similar to what I could see with my bare eyes. I would consider 1, 3 and 5fps like 3 possible level of exposure improvement (and image quality).
Here are the sample pictures:

![01fps](https://user-images.githubusercontent.com/4559307/62012380-595ff500-b17d-11e9-9243-4bd2492b3988.png)
![03fps](https://user-images.githubusercontent.com/4559307/62012382-5e24a900-b17d-11e9-8466-9fa235441343.png)
![05fps](https://user-images.githubusercontent.com/4559307/62012385-611f9980-b17d-11e9-920d-a1187e4d47a9.png)
![10fps](https://user-images.githubusercontent.com/4559307/62012386-6381f380-b17d-11e9-8e7f-fa9477b6467e.png)
![20fps](https://user-images.githubusercontent.com/4559307/62012388-667ce400-b17d-11e9-9d96-81e89c296530.png)
![30fps](https://user-images.githubusercontent.com/4559307/62012390-68df3e00-b17d-11e9-8e3d-39ae9b12fa03.png)
